tests/tests_luks_raid.yml

```diff
@@ -0,0 +1,147 @@
+---
+- hosts: all
+  become: true
+  vars:
+    storage_safe_mode: false
+    mount_location: '/opt/test1'
+    storage_use_partitions: true
+    volume_size: '5g'
+    key_file: '/opt/keyfile'
+    passwd: 'yabbadabbadoo'
+
+  tasks:
+    - include_role:
+        name: storage
+
+    - include_tasks: get_unused_disk.yml
+      vars:
+        min_size: "{{ volume_size }}"
+        disks_needed: 2
+
+    ##
+    ## Raid
+    ##
+
+    - name: Test for correct handling of new encrypted pool w/ no key
+      block:
+        - name: Create an encrypted raid volume
+          include_role:
+            name: storage
+          vars:
+            safe_mode: false
+            storage_volumes:
+              - name: foo
+                type: raid
+                disks: "{{ unused_disks }}"
+                raid_level: "raid0"
+                encryption: true
+                volumes:
+                  - name: test1
+                    mount_point: "{{ mount_location }}"
+                    size: 4g
+
+        - name: unreachable task
+          fail:
+            msg: UNREACH
+
+      rescue:
+        - name: Check that we failed in the role
+          assert:
+            that:
+              - ansible_failed_result.msg != 'UNREACH'
+            msg: "Role has not failed when it should have"
+
+        - name: Verify the output of the keyless luks test
+          assert:
+            that: "blivet_output.failed and
+                   blivet_output.msg|regex_search('encrypted volume.*missing key') and
+                   not blivet_output.changed"
+            msg: "Unexpected behavior w/ encrypted raid pool w/ no key"
+
+    - name: Create an encrypted raid volume w/ default fs
+      include_role:
+        name: storage
+      vars:
+        safe_mode: false
+        storage_volumes:
+          - name: foo
+            type: raid
+            raid_level: "raid0"
+            disks: "{{ unused_disks }}"
+            encryption: true
+            password: "{{ passwd }}"
+            encryption_cipher: 'twofish-xts-plain64'
+            encryption_key_size: 512
+            encryption_luks_version: 'luks2'
+            volumes:
+              - name: test1
+                mount_point: "{{ mount_location }}"
+                size: 4g
+
+    - include_tasks: verify-role-results.yml
+
+    - name: Remove the encryption layer
+      include_role:
+        name: storage
+      vars:
+        safe_mode: false
+        storage_volumes:
+          - name: foo
+            type: raid
+            raid_level: "raid0"
+            disks: "{{ unused_disks }}"
+            encryption: false
+            password: "{{ passwd }}"
+            volumes:
+              - name: test1
+                mount_point: "{{ mount_location }}"
+                size: 4g
+
+    - include_tasks: verify-role-results.yml
+
+    - name: Gerneral a keyfile as password
+      lineinfile:
```

Comment:
Use `copy` with `content` instead of `lineinfile` https://docs.ansible.com/ansible/latest/modules/copy_module.html#copy-module